lib/libdvd/Makefile.in

```diff
@@ -3,80 +3,89 @@ ARCH=@ARCH@
 CC=@CC@
 CXX=@CXX@
 
-SYSDIR=@abs_top_srcdir@/system/players/VideoPlayer
-SOS= libdvdnav-$(ARCH).so
-DIRS=	libdvdread \
-	libdvdnav
+ifeq (@CROSS_COMPILING@,yes)
+  PLATFORM = @host_alias@*
+else
+  PLATFORM = native
+endif
 
-WRAPPER=@abs_top_srcdir@/xbmc/cores/DllLoader/exports/wrapper.o
-WRAPPER_MACH_ALIAS=@abs_top_srcdir@/xbmc/cores/DllLoader/exports/wrapper_mach_alias
+SYSDIR = @abs_top_srcdir@/system/players/VideoPlayer
+SOS = libdvdnav-$(ARCH).so libdvdcss-$(ARCH).so
+
+WRAPPER = @abs_top_srcdir@/xbmc/cores/DllLoader/exports/wrapper.o
+WRAPPER_DEF = @abs_top_srcdir@/xbmc/cores/DllLoader/exports/wrapper.def
+WRAPPER_MACH_ALIAS = @abs_top_srcdir@/xbmc/cores/DllLoader/exports/wrapper_mach_alias
 
 ifeq ($(findstring osx,$(ARCH)),osx)
-LDFLAGS +=-bundle -undefined dynamic_lookup -read_only_relocs suppress
+  LDFLAGS +=-bundle -undefined dynamic_lookup -read_only_relocs suppress
 else
-LDFLAGS += -shared -fPIC -rdynamic
+  LDFLAGS += -shared -fPIC -rdynamic
 endif
 
 ifeq ($(ARCH), powerpc-osx)
-	# Add -lbundle1.o for powerpc-osx
-	BUNDLE1_O = -lbundle1.o
-endif
-
-# Allows us to optionally build xbmc with DVDCSS support
-ifeq (@BUILD_DVDCSS@,1)
-  DIRS += libdvdcss
-  SOS += libdvdcss-$(ARCH).so
-  DVDCSS_A = libdvdcss/.libs/libdvdcss.a
-  DVDCSS_O = libdvdcss/src/.libs/*.o
-  DVDNAV_PREREQ = libdvdcss
+  # Add -lbundle1.o for powerpc-osx
+  BUNDLE1_O = -lbundle1.o
 endif
 
-SLIB=$(addprefix $(SYSDIR)/, $(SOS))
+DVDNAV_A = @abs_top_srcdir@/tools/depends/target/libdvdnav/$(PLATFORM)/.libs/libdvdnav.a
+DVDREAD_A = @abs_top_srcdir@/tools/depends/target/libdvdread/$(PLATFORM)/.libs/libdvdread.a
+DVDCSS_A = @abs_top_srcdir@/tools/depends/target/libdvdcss/$(PLATFORM)/.libs/libdvdcss.a
 
-DISTCLEAN_FILES=includes/dvdread/*.h \
-		includes/dvdcss/*.h \
-		libdvdnav/config.h \
-		libdvdread/config.h
+SLIB = $(addprefix $(SYSDIR)/, $(SOS))
 
-.PHONY: $(DIRS) compile
+DISTCLEAN_FILES=libdvdcss/* libdvdnav/* libdvdread/*
 
 all: $(SLIB)
 
 ifeq ($(findstring osx,$(ARCH)), osx)
 
-$(SYSDIR)/libdvdcss-$(ARCH).so:  $($(WRAPPER)) libdvdcss/.libs/libdvdcss.a
+$(SYSDIR)/libdvdcss-$(ARCH).so:  $(WRAPPER) $(DVDCSS_A)
+	[ -d libdvdcss ] || mkdir libdvdcss
+	cd libdvdcss; ar x $(DVDCSS_A)
 	$(CC) $(LDFLAGS) -Wl,-alias_list,$(WRAPPER_MACH_ALIAS) -o $@ \
-		$(WRAPPER) libdvdcss/src/*.o $(BUNDLE1_O)
-
-$(SYSDIR)/libdvdnav-$(ARCH).so: $($(WRAPPER)) $(DVDCSS_A) libdvdread/obj/libdvdread.a libdvdnav/obj/libdvdnav.a
+		$(WRAPPER) libdvdcss/*.o $(BUNDLE1_O)
+
+$(SYSDIR)/libdvdnav-$(ARCH).so: $(WRAPPER) $(DVDNAV_A) $(DVDREAD_A) $(DVDCSS_A)
+	[ -d libdvdread ] || mkdir libdvdread
+	[ -d libdvdnav ] || mkdir libdvdnav
+	[ -d libdvdcss ] || mkdir libdvdcss
+	cd libdvdcss; ar x $(DVDCSS_A)
+	cd libdvdnav; ar x $(DVDNAV_A)
+	cd libdvdread; ar x $(DVDREAD_A)
```

Comment:
well. after moving to depends, aren't those supposed to be linked like -ldvdcss -ldvdnav -lfoo 
why caling into depends from here and then unpacking the archives ?